lib/Doctrine/DBAL/TransactionManager.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace Doctrine\DBAL;
+
+/**
+ * Manages transactions.
+ */
+class TransactionManager
+{
+    const SAVEPOINT_PREFIX = 'DOCTRINE2_SAVEPOINT_';
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    /**
+     * @var \Doctrine\DBAL\Transaction[]
+     */
+    private $activeTransactions = array();
+
+    /**
+     * Whether nested transactions should use savepoints.
+     *
+     * @var boolean
+     */
+    private $nestTransactionsWithSavepoints = false;
+
+    /**
+     * @param \Doctrine\DBAL\Connection $connection
+     */
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @return TransactionBuilder
+     */
+    public function createTransaction()
+    {
+        return new TransactionBuilder($this);
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return \Doctrine\DBAL\Transaction
+     */
+    public function beginTransaction(array $configuration = array())
+    {
+        $this->connection->connect();
+
+        if (isset($configuration[TransactionBuilder::ISOLATION_LEVEL])) {
+            $this->connection->setTransactionIsolation($configuration[TransactionBuilder::ISOLATION_LEVEL]);
+        }
+
+        $logger = $this->connection->getConfiguration()->getSQLLogger();
+
+        if (count($this->activeTransactions) === 0) {
+            $logger && $logger->startQuery('"START TRANSACTION"');
```

Comment:
We should have a null logger by default instead of having to check if there's a logger in every consuming class.

Reply:
*Introduce null object* ;) Fully agree, this is almost a copy/paste from the current code in Connection.